Zv fast track v0.0.4 (#4)
nibrunieAtSi5/riscv-crypto: doc/vector-extra/riscv-crypto-vector-extra-zvkgs.adoc

[[zvkgs,Zvkgs]]
=== `Zvkgs` - Vector-Scalar GCM/GMAC

Instructions to enable the efficient implementation of parallel versions of GHASH~H~ which is used in Galois/Counter Mode (GCM) and
Galois Message Authentication Code (GMAC).

`Zvkgs` depends on `Zvkg`. It extends the existing `vghsh.vv` and `vgmul.vv` instructions with new vector-scalar variants: `vghsh.vs` and `vgmul.vs`.

The instructions inherit the constraints defined in `Zvkg`:

- element group size (EGS) is 4
- data independent execution timing

> - `vl`/`vstart` must be multiples of EGS=4multiple constraints

Author: ```suggestion
- `vl`/`vstart` must be multiples of EGS=4
```

> All of these instructions work on 128-bit element groups comprised of four 32-bit elements.

Author: ```suggestion
All of these instructions work on 128-bit element groups comprised of four 32-bit elements, in element group parlance `EGS=4`, `EGW=128` and the instructions are only defined for `SEW=32`.
```

> To help avoid side-channel timing attacks, these instructions shall be implemented with data-independent timing.

Author: ```suggestion
To help avoid side-channel timing attacks, these instructions shall always be implemented with data-independent timing.
```

|EGW
|Mnemonic
|Instruction

> | 32 | 128 | vghsh.vs | <<insns-vghsh-vs>>
> | 32 | 128 | vgmul.vs | <<insns-vgmul-vs>>

Author: ```suggestion
| 32 | 128 | `vghsh.vs` | <<insns-vghsh-vs>>
| 32 | 128 | `vgmul.vs` | <<insns-vgmul-vs>>
```
